### DISCUSSION TITLE
Is there a way we can verify the csrftoken in the form data.

### DISCUSSION THREAD

**Discussion by ruifan831** [frankie567/starlette-csrf#11] [Q&A]
Since I am using the HTML form default submission, adding the request header is not possible in this case. Is it possible we can verify the token from form data?

**Comment by frankie567** [comment]
I don't think it's possible easily, this is not how this middleware was designed. For this purpose, I suggest you check out [asgi-csrf](https://github.com/simonw/asgi-csrf) which is designed to work this way.
